### PR TITLE
ci: testplan generation cleanup and scope overhaul

### PR DESCRIPTION
**Pull request by nashif**
- MAINTAINERS: more area coverage
- ci: test_plan: cleanup
- ci: test_plan: rename cmsis-dsp tag
- tests: cmsis-dsp: rename tag to match west module name
- ci: add bluetooth header to bluetooth group

### CODE REVIEW

**Comment by XenuIsWatching** [comment]
Shouldn't the folder name tests/lib/cmsis_dsp/ change to tests/lib/cmsis-dsp/ to match the module name as well. 

tests/lib/cmsis_nn has the same issue with tag since those libraries got migrated out of cmsis-core.

**Review by stephanosio — DISMISSED** [review]

**Comment by nashif** [comment]
> Shouldn't the folder name tests/lib/cmsis_dsp/ change to tests/lib/cmsis-dsp/ to match the module name as well.

personally I prefer underscore in directory names. 

This change is not about renaming for the sake of renaming, it is to match the tag to the module name for the purpose enabling automation and better test coverage when a module has changed.

File: scripts/ci/test_plan.py [hunk 1]
```diff
@@ -303,8 +364,6 @@ def find_tests(self):
             if self.platforms:
                 for platform in self.platforms:
                     _options.extend(["-p", platform])
-            else:
-                _options.append("--all")
```
Comment (PerMac) [inline scripts/ci/test_plan.py:307]: this is quite a major change in the functioning beyond cleanup/refactoring. I am fine with it but just pointing out that that it is hard to catch and nowhere documented.
Before, when a test was changed, a CI was supposed to run on all available platforms. Now those will be only the default ones. IMO integration platforms (and if not added then defautl ones) could make more sense

**Comment by PerMac** [comment]
I think something is still not correct. With the resolved_files added recently, the script keeps track of what was already resolved (scope found) and what not.
 Example: 2 files were modified: a file from a sample and some file in arch. Without this global addition, the script would find the related sample, add it to the scope and marked as resolved. The same with arch. I.e. the scope will be the given sample and this arch. 
With this global, the behavior can be unpredictable. With the above example. If in modified_files "arch" is first and "sample" second: arch will be resolved and the function will exit on sample due to _global. However, the sample was already resolved before. During find_exlcudes the script will see that there are no unresolved files and full scope is not needed. The scope will be sample + arch. If in modified_files the sample is first, the find_arch will exit on it, without resolving the arch change. Hence, during find_excludes, the full scope will be requested.
If you want the behaviour, that if arch and anything else is changed a full scope is needed, you might better set self.full_twister=true before exiting find_arch()

**Pull request closed without merging**